pyccel/ast/builtins.py

```diff
@@ -288,7 +300,26 @@ def is_homogeneous(self):
     def set_arg_types(self,d_vars):
```

Comment:
Can you explain what `d_vars` is here? Maybe with a docstring, or at least with a comment. This will make it much easier to maintain the code later.